packages/@ourworldindata/grapher/src/lineLegend/LineLegend.tsx

```diff
@@ -360,49 +372,160 @@ export class LineLegend extends React.Component<{
+                if (n >= lastBracketValue) return [lastBracketValue, undefined]
+
+                for (const bracket of sortedBrackets) {
+                    if (n >= bracket[0] && n < bracket[1]) return bracket
+                }
+
+                return [undefined, undefined]
+            }
+
+            const sortedCandidates = sortBy(this.initialSeries, (c) => c.midY)
+
+            // pick two candidates, one from the top and one from the bottom
+            const midIndex = Math.floor((sortedCandidates.length - 1) / 2)
+            for (let startIndex = 0; startIndex <= midIndex; startIndex++) {
+                const endIndex = sortedCandidates.length - 1 - startIndex
+                maybePickCandidate(sortedCandidates[endIndex])
+                if (sortedKeepSeries.length >= 2 || startIndex === endIndex)
+                    break
+                maybePickCandidate(sortedCandidates[startIndex])
+                if (sortedKeepSeries.length >= 2) break
+            }
+
+            while (candidates.size > 0 && keepSeriesHeight <= availableHeight) {
+                const sortedBrackets = sortedKeepSeries
+                    .slice(0, -1)
+                    .map((s, i) => [s.midY, sortedKeepSeries[i + 1].midY])
+                    .filter((bracket) => bracket[0] !== bracket[1]) as Bracket[]
+
+                // score each candidate based on how well it fits into the available space
+                const candidateScores: [PlacedSeries, number][] = Array.from(
+                    candidates
+                ).map((candidate) => {
+                    // find the bracket that the candidate is contained in
+                    const [start, end] = findBracket(
+                        sortedBrackets,
+                        candidate.midY
+                    )
+                    // if no bracket is found, return the worst possible score
+                    if (end === undefined || start === undefined)
+                        return [candidate, 0]
+
+                    // score the candidate based on how far it is from the
+                    // middle of the bracket and how large the bracket is
+                    const length = end - start
+                    const midPoint = start + length / 2
+                    const distanceFromMidPoint = Math.abs(
+                        candidate.midY - midPoint
+                    )
+                    const score = length - distanceFromMidPoint
+
+                    return [candidate, score]
+                })
+
+                // pick the candidate with the highest score
+                const maxCandidate = maxBy(candidateScores, (s) => s[1])![0]
+                const picked = maybePickCandidate(maxCandidate)
+
+                // if the highest scoring candidate doesn't fit,
+                // remove it from the candidates and continue
+                if (!picked) candidates.delete(maxCandidate)
```

Comment:
Actually, there's room for another perf improvement here:
```suggestion
                let picked = false
                
                while (!picked && candidateScores.length) {
                    const maxCandidateArr = maxBy(candidateScores, (s) => s[1])
                    const maxCandidate = maxCandidateArr[0]
                    picked = maybePickCandidate(maxCandidate)

                    // if the highest scoring candidate doesn't fit,
                    // remove it from the candidates and continue
                    if (!picked) {
                        candidates.delete(maxCandidate)
                        candidateScores.delete(maxCandidateArr)
                    }
                }
```

... because the scores and brackets don't change until we pick a candidate.